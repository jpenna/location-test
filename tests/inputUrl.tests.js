/* eslint import/no-extraneous-dependencies: "off" */
/*  eslint import/no-unresolved: "off" */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InputUrl from 'src/components/inputUrl';

describe('<InputUrl />', () => {
  let shallowed;
  let handleSubmit;

  beforeAll(() => {
    handleSubmit = jest.fn();
    shallowed = shallow(<InputUrl onSubmit={handleSubmit} />);
  });

  afterEach(() => {
    shallowed.setState({ url: '' });
    jest.clearAllMocks();
  });

  it('should have form, input and a button', () => {
    expect(shallowed.find('form').length).toEqual(1);
    expect(shallowed.find('input').length).toEqual(1);
    expect(shallowed.find('button').length).toEqual(1);
  });

  it('should submit url', () => {
    const url = 'url.com';
    const input = shallowed.find('input');
    input.simulate('change', { target: { value: url } });
    shallowed.find('form').simulate('submit', { preventDefault: () => {} });
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('input should validate url', () => {
    const pattern = shallowed.find('input').prop('pattern');
    const regex = new RegExp(pattern);
    expect(regex.test('url')).toBeFalsy();
    expect(regex.test('http://url.com')).toBeFalsy();
    expect(regex.test('')).toBeFalsy();
  });

  it('should render correctly', () => {
    const rendered = renderer.create(<InputUrl onSubmit={handleSubmit} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
