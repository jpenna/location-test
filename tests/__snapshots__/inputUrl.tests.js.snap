// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputUrl /> should render correctly 1`] = `
<div
  className="content"
>
  <form
    className="has-text-centered"
    onSubmit={[Function]}
  >
    <div
      className="field has-addons has-addons-centered"
    >
      <div
        className="control"
      >
        <input
          className="input"
          onChange={[Function]}
          pattern="^[-a-zA-Z0-9@:%._\\\\+~#=]{2,256}\\\\.[a-z]{2,6}\\\\b([-a-zA-Z0-9@:%_\\\\+.~#?&=]*)"
          placeholder="website.com"
          type="text"
          value=""
        />
        <div
          className="help is-danger show-invalid"
        >
          Format: www.google.com / google.com
        </div>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-primary"
          type="submit"
        >
          Locate!
        </button>
      </div>
    </div>
  </form>
</div>
`;
